Tools/TestWebKitAPI/Tests/WTF/cf/RetainPtr.cpp

```diff
@@ -181,4 +181,18 @@ TEST(RetainPtr, RetainPtrCF)
     EXPECT_EQ(1, CFGetRetainCount(object2.get()));
 }
 
+static void functionWithRetainedCFOutParam(CFNumberRef* __nonnull CF_RETURNS_RETAINED retainedOutParam)
+{
+    float value = 3.1415926535;
+    CFNumberRef number = CFNumberCreate(kCFAllocatorDefault, kCFNumberFloatType, &value);
+    *retainedOutParam = number;
+}
+
+TEST(RetainPtr, RetainedOutParams)
+{
+    RetainPtr<CFNumberRef> value;
+    functionWithRetainedCFOutParam(adoptOutParam(value));
```

Comment:
This idiom is good, calling a function to make adoption of an out parameter work. And I like the name `adoptOutParameter`. Not sure I like the `param` abbreviation but I also wouldn’t like a super-long function name.

Ideally we’d want this to work for both Objective-C pointers and `CFTypeRef` and this test covers only `CFTypeRef`. For the other adopt function, we ended up using two separately named functions, `adoptCF` and `adoptNS`, rather than automatically trying to figure out the flavor. But maybe in this case we can do it with only one, since `RetainPtr` already had to figure out the flavor.

If we did have an Objective-C version, we would want one that does the right thing both without ARC and without ARC.